### PR TITLE
[serving] Adds SageMaker MME support

### PR DESCRIPTION
**Pull request by frankfliu**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: serving/src/main/java/ai/djl/serving/http/ManagementRequestHandler.java [hunk 1]
```diff
@@ -242,40 +213,43 @@ private void handleRegisterModel(final ChannelHandlerContext ctx, QueryStringDec
...
                                     return null;
                                 });
-        if (synchronous) {
-            final String msg = "Model \"" + modelName + "\" registered.";
+        if (req.isSynchronous()) {
+            final String msg = "Model \"" + req.getModelName() + "\" registered.";
             f.thenAccept(v -> NettyUtils.sendJsonResponse(ctx, new StatusResponse(msg)));
         } else {
-            String msg = "Model \"" + modelName + "\" registration scheduled.";
+            String msg = "Model \"" + req.getModelName() + "\" registration scheduled.";
```
Comment (siddvenk) [inline serving/src/main/java/ai/djl/serving/http/ManagementRequestHandler.java:235]: Is msg supposed to be final in one block and not the other?

File: serving/src/main/java/ai/djl/serving/http/InferenceRequestHandler.java [hunk 2]
```diff
@@ -56,6 +56,7 @@ public class InferenceRequestHandler extends HttpRequestHandler {
 
     private static final Pattern PATTERN =
```
Comment (siddvenk) [inline serving/src/main/java/ai/djl/serving/http/InferenceRequestHandler.java:57]: Can we update this variable name to be more clear now that we're introducing INVOKE_PATTERN? Or is it possible to combine these into a single pattern?

File: serving/src/main/java/ai/djl/serving/http/ManagementRequestHandler.java (hunk 1 above)
Comment (frankfliu) [reply serving/src/main/java/ai/djl/serving/http/ManagementRequestHandler.java:235]: When a local variable used in inner class (Lambda), it must be final (or effective final). We explicitly add final to indicate it's being used in inner class

File: serving/src/main/java/ai/djl/serving/http/InferenceRequestHandler.java (hunk 2 above)
Comment (frankfliu) [reply serving/src/main/java/ai/djl/serving/http/InferenceRequestHandler.java:57]: done

**Review by siddvenk — APPROVED** [review]

**Pull request merged**